### PR TITLE
feat(password_input): add formz to password inputs

### LINKED ISSUE #24: Form validation with formz.dart
User validation from [formz.dart](https://pub.dev/packages/formz) model in create wallet forms.

### PR DESCRIPTION
**Pull request by gabaldon**
- [x] Add formz validation to password inputs
- [x] Add formz validation to wallet name input
- [x] Add formz validation to select password form
- [x] Add formz validation for create vtt forms
- [x] Add formz validation for validating an xprv
- [x] Avoid validation on change, instead, add param to allow validation

close #24

### CODE REVIEW

File: lib/screens/create_wallet/enc_xprv_card.dart [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import 'dart:developer';
+
```
Comment (Tommytrg) [inline lib/screens/create_wallet/enc_xprv_card.dart:2]:
```suggestion
```

File: lib/screens/login/view/login_screen.dart [hunk 2]
```diff
@@ -133,49 +133,50 @@ class LoginScreenState extends State<LoginScreen>
...
   }
 
-  String? _validatePassword(String? input) {
-    if (input != null) {
-      if (input.isEmpty) {
-        return 'Please input a password';
-      }
-      return null;
+  bool formValidation() {
+    return _password.valid;
```
[the remark this answers is not in the document]
Comment (gabaldon) [reply lib/screens/login/view/login_screen.dart:137]: we use `Formz.validate` for more than one input

File: lib/widgets/validations/vtt_amount_input.dart [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        try {
+          if (num.parse(value) == 0)
+            return validationUtils.getErrorText(AmountInputError.zero);
+        } catch (e) {
+          return validationUtils.getErrorText(AmountInputError.invalid);
+        }
+      }
+    }
+    return null;
+  }
```
Comment (Tommytrg) [inline lib/widgets/validations/vtt_amount_input.dart:105]: This is a bit hard to read. Could you remove one indentation by changing the first if condition and also adding some comments on each validation block explaining what you are checking in each?

```suggestion
  String? validator(String value, {bool avoidWeightedAmountCheck = false}) {
    final validationUtils = ValidationUtils(errorMap: errorMap);

    if (!this.allowValidation) {
      return null;
    }
    if (value.isEmpty)
      return validationUtils.getErrorText(AmountInputError.empty);
    if (RegExp(r'[a-zA-Z]').hasMatch(value))
      return validationUtils.getErrorText(AmountInputError.invalid);
    if (value.contains('.')) {
      if (value.split('.').length != 2)
        return validationUtils.getErrorText(AmountInputError.invalid);
      if (value.split('.')[1].isEmpty)
        return validationUtils.getErrorText(AmountInputError.invalid);
      if (!RegExp(r'^[0-9]+(\.[0-9]{1,9})?$').hasMatch(value))
        return validationUtils.getErrorText(AmountInputError.decimals);
    }
    if (!RegExp(r'^[0-9]{1,10}(\.[0-9]+)?$').hasMatch(value))
      return validationUtils.getErrorText(AmountInputError.tooBig);
    if (_notEnoughFunds(avoidWeightedAmountCheck: avoidWeightedAmountCheck))
      return validationUtils.getErrorText(AmountInputError.notEnough);
    if (value.isNotEmpty && !this.allowZero) {
      try {
        if (num.parse(value) == 0)
          return validationUtils.getErrorText(AmountInputError.zero);
      } catch (e) {
        return validationUtils.getErrorText(AmountInputError.invalid);
      }
    }
  }
```

File: lib/widgets/validations/xprv_input.dart [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      : super.dirty(value);
+
+  final String? decriptedXprv;
+  final bool allowValidation;
+  final CreateWalletType xprvType;
+
+  @override
+  String? validator(String? value) {
+    final validationUtils = ValidationUtils(errorMap: errorMap);
+    if (this.allowValidation) {
```
Comment (Tommytrg) [inline lib/widgets/validations/xprv_input.dart:34]:
```suggestion
   if (!this.allowValidation) {
     return null;
   }
```

File: lib/widgets/validations/vtt_amount_input.dart [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      nanoWitAmount = _witAmountToNanoWitNumber(value);
+    }
+    return balance < nanoWitAmount;
+  }
+
+  // Override validator to handle validating a given input value.
+  @override
+  String? validator(String value, {bool avoidWeightedAmountCheck = false}) {
+    final validationUtils = ValidationUtils(errorMap: errorMap);
+    if (this.allowValidation) {
```
Comment (Tommytrg) [inline lib/widgets/validations/vtt_amount_input.dart:78]:
```suggestion
   if (!this.allowValidation) {
      return null;
    }
```

File: lib/widgets/validations/password_valid_input.dart [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  // Call super.dirty to represent a modified form input.
+  const VerifyPasswordInput.dirty(
+      {this.allowValidation = false, this.decriptedXprv, String value = ''})
+      : super.dirty(value);
+
+  // Override validator to handle validating a given input value.
+  @override
+  String? validator(String value) {
+    final validationUtils = ValidationUtils(errorMap: errorMap);
+    if (this.allowValidation) {
```
Comment (Tommytrg) [inline lib/widgets/validations/password_valid_input.dart:31]:
```suggestion
    if (!this.allowValidation) {
      return null;
    }
```

File: lib/widgets/validations/password_input.dart [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        super.pure('');
+
+  // Call super.dirty to represent a modified form input.
+  const PasswordInput.dirty({this.allowValidation = false, String value = ''})
+      : super.dirty(value);
+
+  // Override validator to handle validating a given input value.
+  @override
+  String? validator(String value) {
+    if (this.allowValidation) {
```
Comment (Tommytrg) [inline lib/widgets/validations/password_input.dart:26]:
```suggestion
    if (!this.allowValidation) {
      return null;
    }
```

File: lib/widgets/validations/confirmed_password.dart [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      {required this.original, String value = '', this.allowValidation = false})
+      : super.dirty(value);
+
+  final PasswordInput original;
+  final bool allowValidation;
+
+  @override
+  String? validator(String? value) {
+    final validationUtils = ValidationUtils(errorMap: errorMap);
+    if (this.allowValidation) {
```
Comment (Tommytrg) [inline lib/widgets/validations/confirmed_password.dart:28]:
```suggestion
    if (!this.allowValidation) {
      return null;
    }
```

File: lib/widgets/validations/address_input.dart [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Call super.dirty to represent a modified form input.
+  const AddressInput.dirty({String value = '', this.allowValidation = false})
+      : super.dirty(value);
+
+  // Override validator to handle validating a given input value.
+  @override
+  String? validator(String? value) {
+    final validationUtils = ValidationUtils(errorMap: errorMap);
+    if (this.allowValidation) {
```
Comment (Tommytrg) [inline lib/widgets/validations/address_input.dart:30]:
```suggestion
    if (!this.allowValidation) {
      return null;
    }
```

**Review by Tommytrg — APPROVED** [review]

**Pull request merged**